packages/lib/src/io-command.ts

```diff
@@ -939,5 +966,298 @@ export abstract class SelectingInputAPICommand<I, L> extends SelectingInputAPICo
 	protected getIdFromUser = stringGetIdFromUser
 	protected translateToId = stringTranslateToId
 }
+
+
+/**
+ * Use this as your base when you need to be able to list or get a single
+ * resource from a list originating from another resource, which itself can be listed
+ *
+ * This class accepts the identifier types of both the primary and secondary lists as the ID and ID2 generic inputs.
+ * In most cases, your ids will be a simple strings and you can use NestedListingOutputAPICommand2 class,
+ * which is a little simpler. If the first ID is not a string but the second ID is, you can use the
+ * NestedListingOutputAPICommand class.
+ */
+export abstract class NestedListingOutputAPICommandBase<ID, ID2, O, L, L2> extends APICommand {
```

Comment:
I know it's longer but would use `NID` and `NL` for these types and names like `nestedPrimaryKeyName` and `nestedIdOrIndex` instead of just the number "2".